day-01/y_part-2.cpp

```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+int main() {
+
+  std::vector<int> left;
+  std::vector<int> right;
+
+  int l, r;
+  while (std::cin >> l >> r) {
+    left.push_back(l);
+    right.push_back(r);
+  }
+
+  std::set<int> unique{left.begin(), left.end()};
+  std::unordered_map<int, int> counts;
+
+  // could probably use int element : unique
+  for (const auto &element : unique) {
+    counts[element] = std::count(right.begin(), right.end(), element);
+  }
```

Comment:
In therms of performance this loop isn't ideal: You need to traverse the entire right for every element in unique. I think this has quadratic complexitiy (two nested linear traverses). If you were to sort right first, a more efficient lookup could be applied.